core/lib/zksync_core/src/api_server/web3/namespaces/eth.rs

```diff
@@ -523,9 +523,7 @@ impl<G: L1GasPriceProvider> EthNamespace<G> {
                     .await
                     .map_err(|err| internal_error(METHOD_NAME, err))
                 {
-                    if main_node_receipt.status == Some(0.into())
-                        && main_node_receipt.block_number.is_none()
-                    {
+                    if main_node_receipt.status == 0.into() {
```

Comment:
Given that we don't return receipts for rejected transactions anymore, forwarding requests to main node doesn't make sense anymore.